rootex/framework/scene.h

```diff
@@ -51,7 +51,7 @@ class Scene
 	Entity m_Entity;
 
 	Scene* m_ParentScene = nullptr;
-	Vector<Ptr<Scene>> m_ChildrenScenes;
+	Unordered_set<Ptr<Scene>> m_ChildrenScenes;
```

Comment:
Why use an unordered set? What are the advantages of using one here?

Reply:
Basically, we needed O(1) addressing along with O(1) deletion. Based on this, the most ideal data structure seemed to be an unodered_set. O(1) deletion was needed because currently in many functions we loop over the complete vector which isn't necessary.

Reply:
@Bashar-Ahmed Is there any performance testing done to see if using a cache inefficient data structure but with a theoretically better complexity is performing better than a vector?

Also, do you want to save the order in which the children are added in a parent? Would it be confusing in the UI to see the children get reorganized if the map rehashes?

Reply:
No, as of now no performance testing has been done. Will have to look into it and get back to you. Also caching due to vector would've been minimal, if any, due to the fact that it was a vector of ptr, instead of the actual scenes. So when fetching the address,caching would be useful but when getting the object the caching wouldn't be effective. (Had a discussion regarding this with @sin3point14 ). Apologies if this approach is flawed, and do correct me. 

As far as the order is concerned, we looked at this aspect and weren't able to find any viable usecase where saving the order of scenes would be necessary

Reply:
One way the order is important is when you need to iterate over the children to display them on the editor. Seeing the order of the children change as you make any changes in the scene hierarchy can get confusing.